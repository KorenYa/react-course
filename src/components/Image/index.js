import React from 'react';
import PropTypes from 'prop-types';
import './image.scss';
const Image = function(props) {
    return <img className='img-responsive' src={props.src} alt={props.alt} />;
};
Image.propTypes = {
    src: PropTypes.string,
    alt: PropTypes.string
};
export default Image;
