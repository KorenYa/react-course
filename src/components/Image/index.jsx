import React from 'react';
import PropTypes from 'prop-types';
import './image.scss';

const Image = (props) => {
  const { src, alt } = props;
  return <img className="img-responsive" src={src} alt={alt} />;
};
Image.propTypes = {
  src: PropTypes.string,
  alt: PropTypes.string,
};
export default Image;
