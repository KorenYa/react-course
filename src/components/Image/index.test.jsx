import React from 'react';
import renderer from 'react-test-renderer';

import Image from '.';

describe('Image', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Image src="image.jpg" alt="default image" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
