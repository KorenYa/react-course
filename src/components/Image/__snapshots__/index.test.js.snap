// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
<img
  alt="default image"
  className="img-responsive"
  src="image.jpg"
/>
`;
